derive/Cargo.toml

```diff
@@ -18,6 +18,6 @@ edition = "2021"
 proc-macro = true
 
 [dependencies]
-syn = "1.0"
+syn = { version = "2.0", features = ["full"] }
```

Comment:
Do we really need "features=full"? Keeping down the dependency list for this crate is important I think given how low level it is

Reply:
> full — Data structures for representing the syntax tree of all valid Rust source code, including items and expressions.

Code did not compile without enabling full feature